sql/analyzer/resolve_subqueries.go

```diff
@@ -128,8 +148,13 @@ func finalizeSubqueriesHelper(ctx *sql.Context, a *Analyzer, node sql.Node, scop
 }
 
 func resolveSubqueriesHelper(ctx *sql.Context, a *Analyzer, node sql.Node, scope *plan.Scope, sel RuleSelector, finalize bool) (sql.Node, transform.TreeIdentity, error) {
-	return transform.Node(node, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+	return transform.NodeWithCtx(node, nil, func(c transform.Context) (sql.Node, transform.TreeIdentity, error) {
+		n := c.Node
 		if sqa, ok := n.(*plan.SubqueryAlias); ok {
+			if parent, ok := c.Parent.(*plan.JoinNode); ok && sqa.IsLateral {
```

Comment:
this and the one below in `FixFieldIndexesForExpressions` might benefit from a comment, seem like they were tricky to isolate